### ISSUE TITLE
Support for running over ttrees

### ISSUE THREAD

**Issue by TC01** [TC01/treemaker#22]
We can make trees from ntuples, but we can't make trees from other trees. This should be added.

Major bottlenecks here:
- We need to be able to find a tree in a file! We could add configuration syntax for "sourceTreeName" or something, but it'd be a lot nicer if we could iterate through a TFile, recursively through TDirectories, and run over all the TTree objects contained within.
- edmDumpEventContent doesn't work on a ttree (I assume, anyway, from the name). So some kind of code is needed to parse a tree, find all the keys, load them into memory, and present a `labels` dictionary-like thing to the treemaker plugins, to preserve compatibility. This shouldn't be too hard.

But the second point has the added complication that the thing in `labels['branch']` won't be a Handle, unlike the thing in `labels['module']['label']`. What I'm trying to say is, we should maybe have a way to clearly indicate we're running over a ntuple or a ttree, or make plugins declare what their target is (default to ntuple, refuse to run if you try to run over a ttree with a ntuple plugin), or something.

**Comment by TC01** [comment]
So, I finally got around to working on this. The ttrees branch has some preliminary support for running over trees, along with an example plugin!

Plugins now have an "input_type = Ntuple" or "input_type = Tree" variable declaration at the top. (If omitted, the loader defaults to Ntuple). The configuration file syntax now has the following (optional) section, and the `treemaker-config` script has been modified to generate it:

```
[input]
input_type = Tree
source_tree_name = B2GTTreeMaker/B2GTree
```

If input_type = Ntuple, the source_tree_name parameter is ignored. If it's Tree, however, then this is the tree that actually gets processed (obviously).

I wrote a leaves.py which is similar to labels.py, but for trees. It uses the basis of my [treeviewer](https://raw.githubusercontent.com/TC01/rootscripts/master/treeviewer) script to build a map of a ROOT file; the treemaker then vaildates that the tree you asked to run over actually exists in the ROOT file.

And then... otherwise, everything is almost exactly the same as it used to be for ntuples!

Some more testing needs to occur, and some additional checks are probably needed, but this feature is more or less complete.
